lib/simplekiq/orchestration_job.rb

```diff
@@ -39,7 +39,7 @@ def conditionally_within_parent_batch
     end
 
     def orchestration
-      @orchestration ||= Orchestration.new
+      @orchestration ||= Orchestration.new.with_job_options(**job_options)
```

Comment:
🤔 does this mean every orchestration job will have to define a `job_options` method or else it will crash?

Reply:
also do we need `with_job_options` now or can we just pass it to the initializer?

Reply:
> 🤔 does this mean every orchestration job will have to define a job_options method or else it will crash?

Hm - I could do a default `job_options` method on an `OrchestrationJob` with an empty hash, and then it's up to the user to provide one if they want to specialize. Feedback welcome on this, the API is starting to feel a little indirect.

> do we need with_job_options now or can we just pass the options straight to the initializer?

Sure, that can be an initializer arg

Reply:
OK, I like what you have now, except for maybe the naming of `job_options`, though I don't have an obviously better suggestion... I think the thing is that `job_options` makes it sound like it applies to the OrchestrationJob, when in fact it applies to everything *spawned* by the OrchestrationJob. So maybe `child_job_options`?

Reply:
Sure - in retrospect, I can see we could have also done `child_job_default_args` or something as well, instead of manually passing these identical flags to the child jobs. Out of scope anyway.

Reply:
@eapache-opslevel OK, I've reworked this so `child_job_options` is a class method on `OrchestrationJob`. I thought about keeping the instance-level `child_job_options`, but it depends on passing args through OR setting an instance variable. I think this class-method implementation decouples this a little better.

@sabeerzaman I've also updated the readme to use the class method flow to explain setting orchestration-wide settings.

Reply:
Whoops, ignore my other (now deleted) comment that I wrote before I saw your reply here.

>  I thought about keeping the instance-level child_job_options, but it depends on passing args through OR setting an instance variable. I think this class-method implementation decouples this a little better.
Don't entirely follow this reasoning, can you say a bit more? But class-vs-instance method isn't a huge deal right now so it doesn't need to block getting this in.

Reply:
Sure - treating `child_job_options` as a methodless, property-like arg requires we set the sidekiq args into an instance variable, which couples our use case to this library more than I'd like. And moving it up to a class method signaled stateless-ness in a way I liked.

But I don't feel strongly about class-vs-instance.

Reply:
OK, I know it doesn't matter, but both people who reviewed this said they preferred instance method, so it's an instance method now.